c/authService.c

```diff
@@ -173,16 +198,217 @@ static int serveAuthCheck(HttpService *service, HttpResponse *res) {
+  {
+    snprintf(type, STRING_BUFFER_SIZE, "config");
+    snprintf(pluginID, STRING_BUFFER_SIZE, subUrl[0]);
+    snprintf(scope, STRING_BUFFER_SIZE, subUrl[1]);
+    
+  } else {
+    snprintf(type, STRING_BUFFER_SIZE, "service");
+  }
+}
+
+static int makeProfileName(
+  char *profileName,
+  const int profileNameBufSize,
+  const char *type,
+  const char *productCode, 
+  int instanceID, 
+  const char *pluginID, 
+  const char *rootServiceName, 
+  const char *serviceName,
+  const char *method,
+  const char *scope,
+  char subUrl[SAF_SUB_URL_SIZE][STRING_BUFFER_SIZE]) {
+  if (instanceID == -1) {
+    zowelog(NULL, LOG_COMP_ID_SECURITY, ZOWE_LOG_WARNING,
+           "Broken SAF query. Missing instance ID.\n");
+    return -1;
+  }
+  if (method == NULL) {
+    zowelog(NULL, LOG_COMP_ID_SECURITY, ZOWE_LOG_WARNING,
+           "Broken SAF query. Missing method.\n");
+    return -1;
+  }
+  int pos = 0;
+  if (strcmp(type, "service") == 0) {
+    if (pluginID == NULL) {
+      zowelog(NULL, LOG_COMP_ID_SECURITY, ZOWE_LOG_WARNING,
+       "Broken SAF query. Missing plugin ID.\n");
+      return -1;
+    }
+    if (serviceName == NULL) {
+      zowelog(NULL, LOG_COMP_ID_SECURITY, ZOWE_LOG_WARNING,
+       "Broken SAF query. Missing service name.\n");
+      return -1;
+    }
+    pos = snprintf(profileName, profileNameBufSize, "%s.%d.SVC.%s.%s.%s", productCode, instanceID, pluginID, serviceName, method);
+  } else if (strcmp(type, "config") == 0) {
+    if (pluginID == NULL) {
+      zowelog(NULL, LOG_COMP_ID_SECURITY, ZOWE_LOG_WARNING,
+       "Broken SAF query. Missing plugin ID.\n");
+      return -1;
+    }
+    if (scope == NULL) {
+      zowelog(NULL, LOG_COMP_ID_SECURITY, ZOWE_LOG_WARNING,
+       "Broken SAF query. Missing scope.\n");
+      return -1;
+    }
+    pos = snprintf(profileName, profileNameBufSize, "%s.%d.CFG.%s.%s.%s", productCode, instanceID, pluginID, method, scope); 
+  } else if (strcmp(type, "core") == 0) {
+    if (rootServiceName == NULL) {
+      zowelog(NULL, LOG_COMP_ID_SECURITY, ZOWE_LOG_WARNING,
+       "Broken SAF query. Missing root service name.\n");
+      return -1;
+    }
+    pos = snprintf(profileName, profileNameBufSize, "%s.%d.COR.%s.%s", productCode, instanceID, method, rootServiceName); 
+  } else if (pos < 0) {
+    zowelog(NULL, LOG_COMP_ID_SECURITY, ZOWE_LOG_WARNING,
+     "Internal string encoding error.\n");
+    return -1;
+  }
+  // Child endpoints housed via subUrl
+  int index = 0;
+  while (index < SAF_SUB_URL_SIZE && strcmp(subUrl[index], "") != 0) {
+    if (pos >= profileNameBufSize) {
+      break;
+    }
+    pos += snprintf(profileName + pos, profileNameBufSize - pos, ".%s", subUrl[index]);
+    index++;
+  }
+  if (pos >= profileNameBufSize) {
+    char errMsg[256];
+    snprintf(errMsg, sizeof(errMsg), "Generated SAF query longer than %d\n", profileNameBufSize - 1);
+    zowelog(NULL, LOG_COMP_ID_SECURITY, ZOWE_LOG_WARNING, errMsg);
```

Comment:
Can you use the following instead of writing to `errMsg`?
```c
zowelog(NULL, LOG_COMP_ID_SECURITY, ZOWE_LOG_WARNING,
        "Generated SAF query longer than %d\n", profileNameBufSize - 1);
```